### PR TITLE
Show "unmute" option on videos automatically muted by browser.

### LINKED ISSUE #4325: Show "unmute" option on videos automatically muted by browser
Some users may not notice that the video is muted automatically and the overlay would alert them to this. Youtube does something similar.

### PR DESCRIPTION
**Pull request by infinite-persistence**
## PR Type
- [x] Feature -- Fixes #4325 `Show "unmute" option on videos automatically muted by browser`

## What is the current behavior?
Per 4325:
>_Some users may not notice that the video is muted automatically and the overlay would alert them to this. Youtube does something similar._

## What is the new behavior?
image

## Other information
**Implementation:**
- The code is placed in `VideoJs` instead of `VideoViewer` as we need to control the video itself. It's more self-contained here, rather than trying to pass refs around between parent and child.
(_actually, I don't know how to control the player from the parent. I'd be curious to know how to do it in React as a side discussion_)
- useState cannot be used as it will cause a re-render when the hint it clicked and dismissed. The DOM is used to hide the button.

**Dev testing notes:**
[x] When manually unmute using regular button, the hint should go away.
[ ] When manually unmute using 'm', the hint should go away.
          This is actually #4358 `The mute icon sometimes does not reflect actual state.`.  It's either a videojs bug, or an issue with the redux dispatch.  I'm not sure how to address this corner-case. `player.muted()` is always `true` in this scenario.
[x] Should not appear if video failed to load.
[x] Don't show the button when in full screen (following Twitch)
[x] Test with "autoplay" on and off
[x] Test on floating player
[x] Test with entering via "back" vs. direct entry.
[x] Test with dev:web
[x] When video ends, hide it if still present.
[x] Add proper icon
[x] Hint sometimes doesn't appear on fresh lbry.tv. lbryTV: Doesn't appear when random-autoplay next clip. (fixed by changing to `player.one('play')` instead since `player.on('loadstart')` isn't the right event to use for this.)

### CODE REVIEW

File: ui/component/viewers/videoViewer/internal/videojs.jsx [hunk 1]
```diff
@@ -145,6 +193,11 @@ export default React.memo<Props>(function VideoJs(props: Props) {
 
       player = videojs(el, videoJsOptions, () => {
         if (player) {
+          player.one('play', onInitialPlay);
```
Comment (seanyesmunt) [inline ui/component/viewers/videoViewer/internal/videojs.jsx:196]: I think this needs to be `on`

File: ui/component/viewers/videoViewer/internal/videojs.jsx [hunk 2]
```diff
@@ -93,6 +97,50 @@ export default React.memo<Props>(function VideoJs(props: Props) {
...
+  function showTapToUnmute(newState: boolean) {
+    // Use the DOM to control the state of the button to prevent re-renders.
+    // The button only needs to appear once per session.
+    if (tapToUnmuteRef.current) {
+      const curState = tapToUnmuteRef.current.style.visibility === 'visible';
+      if (newState !== curState) {
+        tapToUnmuteRef.current.style.visibility = newState ? 'visible' : 'hidden';
+      }
+    } else if (isDev) {
+      throw new Error('[videojs.jsx] Empty video ref should not happen');
```
Comment (seanyesmunt) [inline ui/component/viewers/videoViewer/internal/videojs.jsx:111]: Instead of `electron-is-dev`, use `const isProduction = process.env.NODE_ENV === "production"`

File: ui/component/viewers/videoViewer/internal/videojs.jsx (hunk 1 above)
Comment (infinite-persistence) [reply ui/component/viewers/videoViewer/internal/videojs.jsx:196]: We only need to run this one time, so `one` is used.  If `on` is used, the button will re-appear every time "Play" is pressed while sound is muted.

Comment (seanyesmunt) [reply ui/component/viewers/videoViewer/internal/videojs.jsx:196]: Interesting. I didn't know videojs supported that.

I tried several videos that started muted and didn't see the button. Can you double-check that it's working?

Comment (infinite-persistence) [reply ui/component/viewers/videoViewer/internal/videojs.jsx:196]: Yeah, I just rebased to confirm the `one` before a few minutes ago and the button is working.

Comment (infinite-persistence) [reply ui/component/viewers/videoViewer/internal/videojs.jsx:196]: Tried a fresh clone on another linux VM and it's ok too.  Let me try on Windows.

Comment (infinite-persistence) [reply ui/component/viewers/videoViewer/internal/videojs.jsx:195]: Ok, if I clear my Electron settings and never touch the videojs player buttons, then I can replicate the issue.  Once I toggled the mute button, it will work from then on.  See if that works for you to test the other aspects.

This is again due to #4358 not reflecting the actual state of the audio.  Let me dig further.  
By the way, in `handleKeyDown` of `VideoJs`, do you know why the `HTMLVideoElement` was used when the `Player` is available? This is my wild guess on what's causing the mismatch.

Comment (infinite-persistence) [reply ui/component/viewers/videoViewer/internal/videojs.jsx:198]: I think I've figured it out.  The Mute and Volume are independent controls and we need to handle two types of "mute":
- Volume icon is pressed to mute
- Volume slider moved to 0%
Fix commited.
